code/modules/mob/living/carbon/life.dm

```diff
@@ -314,7 +314,10 @@
 /mob/living/carbon/handle_status_effects()
 	..()
 	if(staminaloss)
-		adjustStaminaLoss(-3)
+		if(IsKnockdown() || IsUnconscious())

```

Comment:
this should happen in the tick() proc of the knockdown and unconscious status effects.